utils.sh

```diff
@@ -267,11 +267,12 @@ function sync_repo_and_patch {
     pushd $DEST
 
     DEFAULT_BRANCH=$(git remote show origin | sed -n '/HEAD branch/s/.*: //p')
+    REPO_BRANCH=${REPO_BRANCH:-$DEFAULT_BRANCH}
```

Comment:
This assumes we can set REPO_BRANCH globally, which I don't think will always work - e.g we use this function to clone metal3-dev-env, hive and installer repos, where the required branch may be different?

Instead I think we need to support per-repo variables e.g similar to the suggestion [here](https://github.com/openshift-metal3/dev-scripts/pull/1372#discussion_r838318920)

Then we could do e.g `INSTALLER_REPO_BRANCH` `METAL3_DEV_ENV_REPO_BRANCH` etc